src/App.tsx

```diff
@@ -17,18 +47,37 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                searchValue={searchValue}
+                setFilterBy={setFilterBy}
+                setSearchValue={setSearchValue}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoaded ? (
+                <Loader />
+              ) : (
+                <>
+                  {visibleTodos.length ? (
+                    <TodoList
+                      todos={visibleTodos}
+                      selectedTodo={selectedTodo}
+                      setSelectedTodo={setSelectedTodo}
+                    />
+                  ) : (
+                    <p>{errorMessage}</p>
+                  )}
```

Comment:
you should show loader when loading
show error if has error
show todos if there are no error and you're not loading
also, try to avoid nesting ternary operators